README.md

```diff
@@ -0,0 +1,134 @@
+
+**Benefits of Erdstall**&emsp;<!--
+-->In addition to scaling payments as a layer-2 platform, Erdstall also specifically aims to be the secure asset management layer for non-custodial off-chain marketplaces.
+A marketplace building on top of Erdstall does not have to manage any funds on its own, it only has to match the trading parties, while earning a flat fee per facilitated trade.
+This makes the marketplace more trustworthy, as a cyber-attack on the marketplace cannot result in lost funds.
+As a result, Erdstall allows users to enjoy the security, control, and trust of non-custodial trading, combined with the low cost of off-chain transacting.
+Meanwhile, market operators benefit from fast throughput and low operation costs, innate user-trust, and fast and cheap deposits and withdrawals via Erdstall.
+Another benefit is that content creators can natively mint NFTs off-chain, and the cost of on-chain token creation only happens when the NFT is withdrawn by a user into the Ethereum mainnet.
+Because our protocol is very light-weight, it is easy to integrate into existing solutions.
+
+**Scope of this SDK**&emsp;<!--
+-->This SDK is designed to offer the complete client-side functionality of Erdstall.
+It natively integrates with any ethers.js provider, such as Metamask, as its wallet manager.
+The SDK offers full control, but also packages the most important functionality into an easy-to-use API.
+The SDK is designed such that code snippets that use Erdstall can be written generically to be reused within any site using Erdstall.
+That way, common web components, such as payment/donation buttons, can easily be reused on your own homepage.
+
+## Getting started
+
+To get your Erdstall app started, there are a few steps you need to take.
+
+**Importing Erdstall**&emsp;<!--
+-->First, import the SDK's main source directory:
+
+```ts
+import * as SDK from "erdstall-ts-sdk/src"
+```
+
+**Creating the client object**&emsp;<!--
+-->The SDK is straight-forward to use: everything you need is neatly bundled in the `SDK.Erdstall` interface, which you can instantiate via `SDK.NewClient()`, as follows:
+
+```ts
+// Initialize these fields with values received from your ethers or web3 setup.
+var ethAccount: {address: SDK.ledger.Address, signer: ethers.Signer};
+// Create client object.
+let Erdstall = SDK.NewClient(
+	ethAccount.address,
+	ethAccount.signer,
+	new URL("https://operator.erdstall.dev"));
+// Set up any event listeners needed by the application.
+Erdstall.once("open", () => { myDApp.run(Erdstall); });
+// Establish the client's connection.
+Erdstall.initialize();
+```
+
+**Setting up events**&emsp;<!--
+-->After creation, the Erdstall client is now only partially initialized.
+Now is the time to set up any `SDK.ledger.ErdstallEvent` and `SDK.enclave.EnclaveEvent` handlers using `Erdstall.on()` and `Erdstall.once()`.
+The following `EnclaveEvent` events exist (emitted event types are written in parentheses):
+
+* **`"open"`**:
```

Comment:
See comment above. You should at least use constants for the event names. Even better would be to have a separate handler function for each event type. See [jQuery Events](https://api.jquery.com/category/events/), for example.

Reply:
Typescript has type narrowing, which means strings like that are evaluated at compile time. It is not possible to use this API with non-existent events or similar.

Personally I dislike the usage of these `string`s, but it is common in every other web3-provider/library, so I did not want to break with the flow and keep it as ordinary as possible.

I also plan to use some variation of dependent types such that users of this API, given that they use some form of auto-completion, are given the correct eventhandler signature as soon as an event is specified. But I wanted to do this in the second iteration.

Reply:
These are issues that concern the sdk in general and not this README PR. I agree that string constants are nicer, and specific methods are best. But let's just get a first version of the README up and move these discussions into separate issues.